client/source/components/RR/ReviewList/AddReview/AddReview.jsx

```diff
@@ -0,0 +1,184 @@
+import React, { useState, useContext } from 'react';
+import ProdContext from '../../../../ProdContext.js';
+import { Form, InputSubmit } from '../../../shared/form/FormStyles.js';
+import Characteristic from './Characteristic.jsx';
+import FormInput from '../../../shared/form/FormInput.jsx';
+import FormTextarea from '../../../shared/form/FormTextarea';
+import { postID, traitOptions } from '../../utilities/mappings.js';
+import StarRating from './StarRating.jsx';
+import * as requests from '../../../../utilities/axiosRequests.js';
+
+const AddReview = function AddReview({ onClose }) {
+  const { prodID } = useContext(ProdContext);
+
+  const traits = Object.entries(traitOptions);
+
+  const initialFormInput = {
+    product_id: prodID,
+    rating: 0,
+    summary: '',
+    recommend: null,
+    body: '',
+    name: '',
+    email: '',
+    photos: [],
+    characteristics: {}
+  };
+
+  const [formInput, setFormInput] = useState(initialFormInput);
+
+  const clearForm = () => {
+    setFormInput(initialFormInput);
+  };
+
+  const handleStars = (input) => {
+    setFormInput({
+      ...formInput,
+      rating: input
+    });
+  };
+
+  const handleInputChange = (event) => {
+    setFormInput({
+      ...formInput,
+      [event.target.name]: event.target.value
+    });
+  };
+
+  const handleRecommended = (event) => {
+    if (event.target.id === 'yes') {
+      setFormInput({
+        ...formInput,
```

Comment:
Ahh I see what you were saying earlier this week about a destructured object + setting state - here it makes sense bc if you didn't include line 51 your formInput state would be ONLY line 52 and it would erase all the other form input fields